### DISCUSSION TITLE
How to enable zookeeper ACL of Engine?

### DISCUSSION THREAD

**Discussion by wForget** [apache/kyuubi#1202] [Ideas]
#1177 
This PR adds `kyuubi.ha.zookeeper.acl.engine.enabled` configuration to turn off zookeeper ACL of Engine.

Can we specify the kyuubi server keytab file through `--file` so that the spark driver can access this file and turn on the zookeeper ACL of Engine.

**Comment by yaooqinn** [comment]
Yes, I am also thinking of pass it through `--file` to the engine side.

**Comment by yaooqinn** [comment]
would you like to give it a shot @wForget ?

**Reply by wForget** [reply]
Yes, I also hope to support for other zookeeper ACL schemes through the `org.apache.curator.framework.CuratorFrameworkFactory.Builder#authorization` method.

**Reply by yaooqinn** [reply]
sounds good too

**Reply by wForget** [reply]
Thanks, i will try it.

**Reply by cxzl25** [reply]
If the uploaded file is a service keytab, kerberos authentication should fail?
If the uploaded file is the user's keytab, do we need to add many user acl entries to the znode?

**Reply by cxzl25** [reply]
Oh, I just saw the pr you submitted, which supports multiple authentication methods.

**Reply by wForget** [reply]
> Oh, I just saw the pr you submitted, which supports multiple authentication methods.

Yes, using the service's keytab will not fail, and the zookeeper authentication keytab can be configured. In addition, the zookeeper authentication scheme of digest is added. See #1204 for details.
